proto-lens-tests/tests/test_proto3_optional.proto

```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+
+package proto3_optional;
+
+message Foo {
+  int32 untracked = 1;
+  optional int32 tracked = 2;
+  oneof nonsynth {
```

Comment:
I didn't see nonsynth or alternative used in any of the tests; could we remove them? (There are already other tests specifically focused on oneofs.)

Reply:
The compile-time checks I added to guard against extraneous generated API make use of these additional symbols.